pycqed/measurement/pulse_sequences/fluxing_sequences.py

```diff
@@ -559,9 +781,19 @@ def cphase_seqs(qbc_name, qbt_name, hard_sweep_dict, soft_sweep_dict,
     final_rotations[0]['ref_pulse'] = 'cphase_init_pi_qbc'
     final_rotations[0]['pulse_delay'] = delay
 
+    # make sure RO happens after last ge
+    ge_pulse_len = np.array(
+        [final_rotations[i]['sigma']*final_rotations[i]['nr_sigma']
+         for i in range(2)])
+
+    longer_pulse = ge_pulse_len.argmax()
+
+
     ro_pulses = generate_mux_ro_pulse_list([qbc_name, qbt_name],
                                             operation_dict)
 
+    ro_pulses[0]['ref_pulse'] = final_rotations[longer_pulse]['name']
```

Comment:
would be cleaner to use multiref feature (as soon as @antsr has reviewed that other pull request ;-))

Reply:
This works after merging the latest master:

@@ -786,22 +786,16 @@ def cphase_seqs(qbc_name, qbt_name, hard_sweep_dict, soft_sweep_dict,
     #     print(f'delay adjusted to {delay*1e9:.2f} ns '
     #           f'to fulfill commensurability conditions with 16/2.4e9.')
     final_rotations[0]['ref_pulse'] = 'cphase_init_pi_qbc'
     final_rotations[0]['pulse_delay'] = delay
 
-    # make sure RO happens after last ge
-    ge_pulse_len = np.array(
-        [final_rotations[i]['sigma']*final_rotations[i]['nr_sigma']
-         for i in range(2)])
-
-    longer_pulse = ge_pulse_len.argmax()
-
-
     ro_pulses = generate_mux_ro_pulse_list([qbc_name, qbt_name],
                                             operation_dict)
 
-    ro_pulses[0]['ref_pulse'] = final_rotations[longer_pulse]['name']
+    # make sure RO happens after last ge
+    ro_pulses[0]['ref_pulse'] = [p['name'] for p in final_rotations]
 
     hsl = len(list(hard_sweep_dict.values())[0]['values'])
     params = {'cphase_init_pi_qbc.amplitude': np.concatenate(
         [initial_rotations[0]['amplitude']*np.ones(hsl//2), np.zeros(hsl//2)]),
               'cphase_final_pi_qbc.amplitude': np.concatenate(

Reply:
Is now implemented as suggested!